### PR TITLE
ZSH - changed dependancy pcre->pcre2 since pcre is not compiling

### PR DESCRIPTION
**Pull request by Behinder**
###### Type(s)

- [ X] bugfix
- [ ] enhancement
- [ ] security fix

### CODE REVIEW

File: shells/zsh/Portfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ homepage                https://www.zsh.org
 depends_lib             port:gdbm \
                         port:libiconv \
                         port:ncurses \
-                        port:pcre
+                        port:pcre2
```
Comment (reneeotten) [inline shells/zsh/Portfile:24]: Changing a library dependency requires a "revision" increase.

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by reneeotten** [comment]
and your commit message doesn't follow our guidelines. It should be "<portname>: short description"; the port name isn't capitalized either.

File: shells/zsh/Portfile (hunk 1 above)
Comment (jmroot) [reply shells/zsh/Portfile:24]: And also usually some change in configure flags or other way of controlling which libraries are used.

**Comment by reneeotten** [comment]
@Behinder please take care of the review comments and rebase with "masfer" to get rid of the merge conmit

**Comment by ryandesign** [comment]
We are not aware that "pcre is not compiling"; if this is your experience, please file a bug report about it.

File: net/libpsl/Portfile [hunk 2]
```diff
@@ -49,7 +49,7 @@ checksums           ${main_distfile} \
 # See e.g. https://trac.macports.org/ticket/60419
 
 # DO NOT change this unless you have understood and acted on the above comment!
-set py_ver          3.10
+set py_ver          3.12
```
Comment (reneeotten) [inline net/libpsl/Portfile:52]: did you read the comment above and made absolutely sure that nothing would break with this change?

**Review by reneeotten — CHANGES_REQUESTED** [review]
also, you didn't actually address @jmroot his comment about making sure that it will use/link with "pcre2".

**Comment by reneeotten** [comment]
this PR is factually incorrect. If you look what the configure script does, it is looking for `pcre.h` which is a file that only is present in the `pcre` port (in the `pcre2` port it is called `pcre2.h`. So the changes you suggest here will result in `configure` not finding `pcre` and compiling `zsh` without `pcre` support.
You can verify that this is the case yourself as you will see no mentions of `pcre2` when doing:
```
otool -L `port contents zsh | grep  "\.so"` | grep pcre
```
whereas you will find linking to the `pcre` port. 

Also, the `pcre` port is installing just fine here locally and also according to the [status page](https://ports.macports.org/port/pcre/details/) it builds everywhere .

Therefore, closing this PR ; if you still have problems with the `pcre` port please file a Trac ticket.

**Comment by jmroot** [comment]
FTR, it looks like the next upstream release of zsh will add pcre2 support.

**Pull request closed without merging**